### PR TITLE
DP-19354 Search button behavior

### PR DESCRIPTION
**Pull request by ygannett**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-19354)
- [Github issue]()

## Steps to Test

Set the behavior to put focus on the search input in the hamburger menu by clicking the search button on the blue bar when the menu is open.

1. Go to http://localhost:3000/patterns/05-pages-Homepage-hamburger/05-pages-Homepage-hamburger.html.
2. In the window width of 620px and smaller, click the search button on the blue bar and check its behavior:
- When the search button is clicked while the menu is CLOSE, the menu opens and focus is set to the search input box.
- When the search button is clicked while the menu is OPEN, focus is set to the search input box. Anything else doesn't change.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

**Comment by tom-fleming** [comment]
@ygannett Is the intention here that the search field will come into focus so that the keyboard opens up on mobile devices? If so, on iOS at least, the field comes into focus, but it does not open up the keyboard. To accomplish that, I think we would need to look into a suggestion like this one: https://stackoverflow.com/questions/54424729/ios-show-keyboard-on-input-focus

File: patternlab/styleguide/source/assets/js/modules/mainNavHamburger.js [hunk 1]
```diff
@@ -268,24 +263,20 @@ function closeMenu() {
...
-}
-
 function commonCloseMenuTasks() {
   body.classList.remove("show-menu");
 
   if (width < 841) {
     menuButton.removeAttribute("style");
   }
 
-  menuButtonText[0].textContent = "Menu";
```
Comment (ygannett) [inline patternlab/styleguide/source/assets/js/modules/mainNavHamburger.js]: Removed these lines.  The label text change will be handled via css which is done in DP-19337.

**Comment by ygannett** [comment]
@tom-fleming Thank you for mentioning the keyboard display in mobile.  It's implemented following the link.  Can you take a look again?

**Comment by tom-fleming** [comment]
@ygannett Search comes into focus for me, but it doesn't pull up the keyboard.
image

**Comment by ygannett** [comment]
The backstop failure via pattternlab_test is not related to this branch's change.

**Comment by ygannett** [comment]
@tom-fleming Can you review this again?

**Comment by tom-fleming** [comment]
@ygannett It was fun testing and having the keyboard pop up :) I noticed some issues with the search box itself switching between rounded corners and more squared corners, but that might fall out of the scope of this issue. I think this is almost ready, but I merged in changes from DP-17200 and had to resolve some merge conflicts. Could you review this one more time?

**Pull request merged**